src/hotspot/share/prims/vectorSupport.cpp

```diff
@@ -479,6 +479,15 @@ int VectorSupport::vop2ideal(jint id, BasicType bt) {
       }
       break;
     }
+    case VECTOR_OP_POPCNT: {
```

Comment:
Can the fall through ever occur? since the operation is only supported for `int` and `long`.

Reply:
It is also wrong, you can't use the scalar `PopCountI` on byte and short

Reply:
Also, are you sure this will work? Since PopCountVL takes a long vector and returns an int vector, I think you need to adjust the C2 compiler before trying to intrinsify this.